### ISSUE TITLE
Missing FFMPEG error when attempting to transcode with Tiddl

### ISSUE THREAD

**Issue by NotPhoton** [cstaelen/tidarr#34]
Hello!
In my environment variables, I have the file extension set to flac. When Tiddl attempts to transcode a file from m4a to flac, it throws an error about missing the ffmpeg file or directory

> DOWNLOAD_PATH "/home/app/standalone/download/incomplete"
> TRACK_QUALITY "HI_RES_LOSSLESS"
> TRACK_TEMPLATE "{artist}/{album} ({year})/{artist} - {album} - {number} - {title}"
> ALBUM_TEMPLATE "{artist}/{album}/{title}"
> PLAYLIST_TEMPLATE "{playlist}/{title}"
,> FILE_EXTENSION "flac"
,Tiddl: [0;32m[ TIDDL ] [0msaved settings to /root/.tiddl_config.json[0m
,Tiddl: [0;32m[ TIDDL ] [0mSuperheaven - Long Gone - 1 - Long Gone :: Max Quality - 24 bit 96.0 kHz[0m
,threaded download ■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■ 100%,
,Tiddl: Traceback (most recent call last):
  File "/usr/bin/tiddl", line 8, in <module>
,Tiddl:     sys.exit(main())
             ^^^^^^,Tiddl: 
  File "/usr/lib/python3.11/site-packages/tiddl/__init__.py", line 255, in main
    downloadTrack(
,Tiddl:   File "/usr/lib/python3.11/site-packages/tiddl/__init__.py", line 188, in downloadTrack
    track_path = convertToFlac(
                 ,Tiddl: ^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/tiddl/utils.py", line 179, in convertToFlac
,Tiddl:     result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/subprocess.py", line 548, in run
,Tiddl:     with Popen(*popenargs, **kwargs) as process:
,Tiddl:          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/subprocess.py", line 1026, in __init__
,Tiddl:     self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.11/subprocess.py", line 1955, in _execute_child
,Tiddl:     raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'ffmpeg'
,Tiddl process exited with code 1" `

This only happens with songs in m4a format (typically master quality releases). The functionality seems to work fine in a regular install of TIDDL, provided ffmpeg is installed on the device. I did see an issue referencing something similar over at Tiddl's repo
https://github.com/oskvr37/tiddl/issues/52

**Comment by diegocade1** [comment]
having the same error

`threaded download ■■■■■■■■■■■■■■■■■■■■■■■■■■■■■□ 97%

threaded download ■■■■■■■■■■■■■■■■■■■■■■■■■■■■■□ 98%

threaded download ■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■ 100%

Tiddl: Traceback (most recent call last):

  File "/usr/bin/tiddl", line 8, in <module>

Tiddl:     sys.exit(main())

             ^^^^^^

  File "/usr/lib/python3.11/site-packages/tiddl/__init__.py", line 266, in main

    downloadAlbum(input_id, skip_existing)

  File "/usr/lib/python3.11/site-packages/tiddl/__init__.py", line 217, in downloadAlbum

Tiddl:     file_dir, file_name = downloadTrack(

                          ^^^^^^^^^^^^^^

  File "/usr/lib/python3.11/site-packages/tiddl/__init__.py", line 188, in downloadTrack

    track_path = convertToFlac(

                 ^^^^^^^^^^^^^^

  File "/usr/lib/python3.11/site-packages/tiddl/utils.py", line 179, in convertToFlac

Tiddl:     result = subprocess.run(

             ^^^^^^^^^^^^^^^

  File "/usr/lib/python3.11/subprocess.py", line 548, in run

    with Popen(*popenargs, **kwargs) as process:

     

Tiddl:     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/lib/python3.11/subprocess.py", line 1026, in __init__

Tiddl:     self._execute_child(args, executable, preexec_fn, close_fds,

  File "/usr/lib/python3.11/subprocess.py", line 1955, in _execute_child

Tiddl:     raise child_exception_type(errno_num, err_msg, err_filename)

FileNotFoundError: [Errno 2] No such file or directory: 'ffmpeg'

Tiddl process exited with code 1`

**Comment by diegocade1** [comment]
having the same error

```
threaded download ■■■■■■■■■■■■■■■■■■■■■■■■■■■■■□ 97%

threaded download ■■■■■■■■■■■■■■■■■■■■■■■■■■■■■□ 98%

threaded download ■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■ 100%

Tiddl: Traceback (most recent call last):

  File "/usr/bin/tiddl", line 8, in <module>

Tiddl:     sys.exit(main())

             ^^^^^^

  File "/usr/lib/python3.11/site-packages/tiddl/__init__.py", line 266, in main

    downloadAlbum(input_id, skip_existing)

  File "/usr/lib/python3.11/site-packages/tiddl/__init__.py", line 217, in downloadAlbum

Tiddl:     file_dir, file_name = downloadTrack(

                          ^^^^^^^^^^^^^^

  File "/usr/lib/python3.11/site-packages/tiddl/__init__.py", line 188, in downloadTrack

    track_path = convertToFlac(

                 ^^^^^^^^^^^^^^

  File "/usr/lib/python3.11/site-packages/tiddl/utils.py", line 179, in convertToFlac

Tiddl:     result = subprocess.run(

             ^^^^^^^^^^^^^^^

  File "/usr/lib/python3.11/subprocess.py", line 548, in run

    with Popen(*popenargs, **kwargs) as process:

     

Tiddl:     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/lib/python3.11/subprocess.py", line 1026, in __init__

Tiddl:     self._execute_child(args, executable, preexec_fn, close_fds,

  File "/usr/lib/python3.11/subprocess.py", line 1955, in _execute_child

Tiddl:     raise child_exception_type(errno_num, err_msg, err_filename)

FileNotFoundError: [Errno 2] No such file or directory: 'ffmpeg'

Tiddl process exited with code 1
```

**Comment by cstaelen** [comment]
Hi there,

Thanks for your feedback 🙏 

A fix is on the way, stay tuned for next release (0.1.6).

**Comment by cstaelen** [comment]
`0.1.6` is available. You can `docker pull tidarr`. 👍

**Comment by diegocade1** [comment]
thank you @cstaelen

**Comment by diegocade1** [comment]
hey there @cstaelen , do you know if there's an option to let LOSSLESS as the default?

![image](https://github.com/user-attachments/assets/6c05255b-7ebf-41ec-88a4-7a6edd944fcf)

(i know this is not the place to ask question), do you have a subreddit for questions?

**Comment by cstaelen** [comment]
Good suggestion 👍

Maybe i could store the filter value in localStorage of browsers to remember it.  It would be the easiest way with no extra params.

Or another way to achieve that would be to have a new environment var in `docker-compse.yml` to optionnaly force it as default value (cross browser solution).

Or maybe the twice ? 

Wich approch seems the best to you ?

**Comment by diegocade1** [comment]
mmmm to me the second one would be better, to have a new environment var, but honestly, I wouldnt mind one or both.
Just make the one thats easier for you :)

**Comment by cstaelen** [comment]
Hello @diegocade1 , 

I just released 0.1.7 version with new features you may be interested in : 

- Store selected quality filter in browser local storage
- Force default quality filter for all devices with env var : `-  REACT_APP_TIDARR_DEFAULT_QUALITY_FILTER=<"lossless" | "high" | "all">`
- Add an authentication form with env var `- ADMIN_PASSWORD=123pwd`

Hope you will enjoy :)

**Comment by diegocade1** [comment]
@cstaelen you are awesome, thank you!
